### ISSUE TITLE
[Issue]: Remove background failures due to onnxruntime

### ISSUE THREAD

**Issue by mark-wd** [vladmandic/automatic#711]
### Issue Description

As of the latest version, attempting to remove backgrounds with any of the u2net variants is giving me the error below, with or without an upsacling operation involved. I recently turned xformers on, no idea if it is related in any way.

Towards the bottom It also seems to be returning an error in the "D" drive, but well... I _don't have_ a D drive.

```
ERROR    gradio call: RuntimeError
╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ G:\stable-diffusion-vladmandic\modules\call_queue.py:59 in f                                                         │
│                                                                                                                      │
│    58 │   │   │   │   pr.enable()                                                                                    │
│ ❱  59 │   │   │   res = list(func(*args, **kwargs))                                                                  │
│    60 │   │   │   if shared.cmd_opts.profile:                                                                        │
│                                                                                                                      │
│ G:\stable-diffusion-vladmandic\modules\call_queue.py:38 in f                                                         │
│                                                                                                                      │
│    37 │   │   │   try:                                                                                               │
│ ❱  38 │   │   │   │   res = func(*args, **kwargs)                                                                    │
│    39 │   │   │   │   progress.record_results(id_task, res)                                                          │
│                                                                                                                      │
│                                               ... 4 frames hidden ...                                                │
│                                                                                                                      │
│ G:\stable-diffusion-vladmandic\venv\lib\site-packages\onnxruntime\capi\onnxruntime_inference_collection.py:360 in    │
│ __init__                                                                                                             │
│                                                                                                                      │
│   359 │   │   try:                                                                                                   │
│ ❱ 360 │   │   │   self._create_inference_session(providers, provider_options, disabled_optimiz                       │
│   361 │   │   except ValueError:                                                                                     │
│                                                                                                                      │
│ G:\stable-diffusion-vladmandic\venv\lib\site-packages\onnxruntime\capi\onnxruntime_inference_collection.py:408 in    │
│ _create_inference_session                                                                                            │
│                                                                                                                      │
│   407 │   │   # initialize the C++ InferenceSession                                                                  │
│ ❱ 408 │   │   sess.initialize_session(providers, provider_options, disabled_optimizers)                              │
│   409                                                                                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
RuntimeError: D:\a\_work\1\s\onnxruntime\core\session\provider_bridge_ort.cc:1106 onnxruntime::ProviderLibrary::Get
[ONNXRuntimeError] : 1 : FAIL : LoadLibrary failed with error 126 "" when trying to load
"G:\stable-diffusion-vladmandic\venv\lib\site-packages\onnxruntime\capi\onnxruntime_providers_tensorrt.dll"
```

### Version Platform Description

2023-05-02 hash: eb03fce3

**Comment by vladmandic** [comment]
D:\xxx is the original location of the source code of the file which generated the error when `onnxruntime` was compiled. thats a braindead way of logging, usually done back in 80ies, but whatever.

anyhow, issue is with `onnxruntime`. but when you say "attempting to remove backgrounds", i don't really know what exactly do you mean - can you document exact workflow you're attempting?

**Comment by mark-wd** [comment]
Sure.

Process Image > Drop image into box > Set resize to "1" > Upscaler "None" > Remove background "u2net" -> Return mask and Alpha matting set to "unchecked"

The process animation is shown on the right box, fails, and the error pops up in the log.

**Comment by mark-wd** [comment]
Doing a fresh install fixed the problem. Maybe recent updates broke something.

**Comment by mark-wd** [comment]
Nope. Worked for a moment and then giving errors again: [imgur](https://imgur.com/a/KcSTLd5)

**Comment by vladmandic** [comment]
can you check: <https://github.com/danielgatis/rembg/issues/312> and <https://github.com/AUTOMATIC1111/stable-diffusion-webui-rembg/issues/3>

it seems that `onnxruntime` library is a bit fussy.

**Comment by mark-wd** [comment]
I tried following the steps, but still getting errors. The strange thing is that this is working just fine on auto1111 with the extension bolted on.

**Comment by vladmandic** [comment]
what's your onnxruntime version both here and original a1111 repo?

> venv\Scripts\activate
> pip show onnxruntime

**Comment by mark-wd** [comment]
Scratch that. Fresh installation of 1111 also failing now. I'm unable to say if it was a change to my system or a PR that was merged into both projects. Any idea on how to move forward with troubleshooting? Might be useful to other users who experience this problem now or in the future. Maybe there's also something that can be enforced from the webui side?

Library info:

Name: onnxruntime
Version: 1.14.1
Summary: ONNX Runtime is a runtime accelerator for Machine Learning models
Home-page: https://onnxruntime.ai
Author: Microsoft Corporation
Author-email: onnxruntime@microsoft.com
License: MIT License
Location: g:\stable-diffusion-vladmandic\venv\lib\site-packages
Requires: coloredlogs, flatbuffers, numpy, packaging, protobuf, sympy
Required-by: rembg

**Comment by vladmandic** [comment]
two questions
- what is storage behind `G:\stable-diffusion-vladmandic`
  G: is quite high on alphabet, is that a network mount or something like that?
- what is the output of `dir venv\lib\site-packages\onnxruntime\capi`

**Comment by tonny1314sy** [comment]
https://github.com/AUTOMATIC1111/stable-diffusion-webui-rembg/pull/14
Hello, it seems that a temporary solution has been found, but it has not been adopted

**Comment by mark-wd** [comment]
Any chance that you could merge the above, @vladmandic?

**Comment by vladmandic** [comment]
if i merge, then copy gets out-of-sync with original and if there are any future upgrades, they wouldn't be applicable.
i'll give author one more week to merge, if not i'll have to take it and maintain copy with all the fixes.

**Comment by mark-wd** [comment]
Makes sense. Will wait and watch.

**Comment by mark-wd** [comment]
Brief update: manually "pip install onnxruntime==1.14.0" really does fix the problem.

**Comment by vladmandic** [comment]
done. hope it fixes it, if not post here and i'll reopen.
